nmodl/ode.py

```diff
@@ -291,7 +293,19 @@ def solve_non_lin_system(eq_strings, vars, constants, function_calls):
     # interweave
     code = _interweave_eqs(vecFcode, vecJcode)
 
-    return code
+    return code, linear
+
+
+def _is_linear(eqs, state_vars, sympy_vars):
+    for expr in eqs:
+        for x in state_vars:
+            for y in state_vars:
```

Comment:
Could we here use: https://docs.python.org/3/library/itertools.html#itertools.combinations instead to increase speed?

Reply:
Yes, I can do that.
Then it would only check one of the derivatives: "dxdy", "dydx" but not both.

Reply:
done